### PR TITLE
fix  no defined OPENSSL_THREADS error on static build 

### PR DESCRIPTION
**Pull request by jingjingxyk**
fix error   on Static build

image

coturn static build command : 

```bash 
export TURN_NO_PROMETHEUS=ON
export TURN_NO_SYSTEMD=ON
export TURN_NO_MYSQL=ON
export TURN_NO_MONGO=ON
export TURN_NO_SQLITE=OFF
export TURN_NO_PQ=OFF
export TURN_NO_HIREDIS=OFF


PACKAGES='sqlite3'
PACKAGES="\$PACKAGES libevent  libevent_core libevent_extra  libevent_openssl  libevent_pthreads"
PACKAGES="\$PACKAGES libpq"
PACKAGES="\$PACKAGES hiredis"
SSL_CFLAGS="$(pkg-config  --cflags-only-I  --static openssl libcrypto libssl) " \
SSL_LIBS="$(pkg-config    --libs           --static openssl libcrypto libssl) " \
CPPFLAGS="$(pkg-config  --cflags-only-I  --static \$PACKAGES)" \
LDFLAGS="$(pkg-config   --libs-only-L    --static \$PACKAGES) -static" \
LIBS="$(pkg-config      --libs-only-l    --static \$PACKAGES) -lstdc++ -lm -lpgcommon -lpgport " \
CFLAGS="-O3  -g  -std=gnu11 " \
./configure  \
--prefix=/opt/coturn/
make -j $(nproc)
make install
```

### CODE REVIEW

**Comment by KangLin** [comment]
Please use C11.

**Comment by ggarber** [comment]
Thanks for working on this @jingjingxyk   Let me know when you think it is ready for review

**Comment by jingjingxyk** [comment]
ext-dependency-graph

**Comment by jingjingxyk** [comment]
@ggarber  Now, Verifying .

**Comment by jingjingxyk** [comment]
image

**Comment by Neustradamus** [comment]
@jingjingxyk: How have you done your graph?

**Comment by jingjingxyk** [comment]
> @jingjingxyk: How have you done your graph?

with graphviz  and  [jingjingxyk/swoole_cli](https://github.com/jingjingxyk/swoole-cli/blob/new_dev/sapi/src/template/extension-dependency-graph.php)   ; project branch [`new_dev`](https://github.com/jingjingxyk/swoole-cli/blob/new_dev/)

[coturn static build config ](https://github.com/jingjingxyk/swoole-cli/blob/04d305b8c84908b59434063d30264f7cc1219f5d/sapi/src/builder/library/coturn.php#L45)

[generate dependency graph docs](https://github.com/jingjingxyk/swoole-cli/blob/new_dev/docs/options.md#with-dependency-graph)

```bash

git clone -b new_dev --depth=1  https://github.com/jingjingxyk/swoole-cli.git
cd swoole_cli
sh setup-php-runtime.sh 
composer update 


php prepare.php  +coturn  --without-docker=1 --skip-download=1 --with-dependency-graph=1

bash sapi/extension-dependency-graph/generate-dependency-graph.sh


ls -lh bin/ext-dependency-graph.*


```

image

File: src/apps/relay/mainrelay.c [hunk 1]
```diff
@@ -3286,13 +3288,12 @@ int THREAD_cleanup(void) {
   mutex_buf_initialized = 0;
   return 1;
 }
+/* OPENSSL_VERSION_NUMBER < OPENSSL_VERSION_1_1_0 */
+/* defined(OPENSSL_THREADS) end */
```
Comment (jonesmz) [inline src/apps/relay/mainrelay.c:3292]: The #if-checks were combined into a single check.

Maybe these two lines of comments should be replaced with

/* #if defined(OPENSSL_THREADS) && (OPENSSL_VERSION_NUMBER < OPENSSL_VERSION_1_1_0) */

for clarity?

**Review by jonesmz — APPROVED** [review]

**Comment by jingjingxyk** [comment]
Issue resolved

```shell
export CFLAGS="${CFLAGS} -DOPENSSL_THREADS  "

```

File: src/apps/relay/mainrelay.c (hunk 1 above)
Comment (jingjingxyk) [reply src/apps/relay/mainrelay.c]: Issue resolved . use  ` -DOPENSSL_THREADS`
```shell
export CFLAGS="${CFLAGS} -DOPENSSL_THREADS  "
```

**Pull request closed without merging**